portfolio/src/main/webapp/script.js

```diff
@@ -185,9 +201,146 @@ function prevComments(){
     });
 }
 
+/**
+ * 
+ * @param {*} authData json authentication data fetched from server
+ * 
+ * Based on server response it creates AuthData object that contains login state
+ * LoginState is evaluated based on response:
+ * -if user is not logged in LoginState if obviously LOGGED_OUT
+ * -if user is logged in and the nickname was not contained in response than 
+ *     it means user haven't set nickname yet hence the loginState is LOGGED_IN_NICKNAME_NOT_SET
+ * -if user is logged in and the nickname was send back from server than state is LOGGED_IN_NICKNAME_SET
+ */
+function handleAuthData(authData){
+    let loginState = LoginState.LOGGED_OUT;
+    const nickname = authData.nickname;
+    const userId = authData.userId;
+    const loginUrl = authData.loginUrl;
+    const logoutUrl = authData.logoutUrl;
+
+    if(typeof authData.isUserLoggedIn !== 'undefined' && authData.isUserLoggedIn == false){
```

Comment:
Instead of `typeof`, can we use `'isUserLoggedIn' in authData`.

Reply:
I'm not sure if i understand. First we checking if such parameter isUserLoggedIn (using typeof) was provided in server response, than we check it's value.

Reply:
`if ('isUerLoggedIn' in authData)` this is the same check as using `typeof` but more readable.